modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/internal/util/ItemSelectorRepositoryEntryBrowserUtil.java

```diff
@@ -261,4 +255,48 @@ private static JSONObject _createJSONObject(String key, String value) {
 		);
 	}
 
+	private static Folder _getFolder(long folderId) throws Exception {
+		Folder folder = null;
+
+		if (folderId != DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+			folder = DLAppServiceUtil.getFolder(folderId);
+		}
+
+		return folder;
+	}
+
+	private static Group _getGroup(
+			Folder folder, HttpServletRequest httpServletRequest,
+			long repositoryId)
+		throws Exception {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		Group group = themeDisplay.getScopeGroup();
+
+		if (repositoryId != 0) {
+			try {
+				group = GroupServiceUtil.getGroup(repositoryId);
+			}
+			catch (PortalException portalException) {
+				_log.error(portalException);
+
+				Repository repository = RepositoryServiceUtil.getRepository(
+					repositoryId);
+
+				group = GroupServiceUtil.getGroup(repository.getGroupId());
+			}
+		}
+		else if (folder != null) {
+			group = GroupServiceUtil.getGroup(folder.getGroupId());
+		}
+
+		return group;
```

Comment:
Also, instead of updating the `group` variable like this, can you use direct returns?
```
if (repositoryId != 0) {
    try {
         return GroupServiceUtil.getGroup(repositoryId);
    }
    catch ( ...) {
        ....
        return GroupServiceUtil.getGroup(repository.getGroupId());
    }
}
else if (folder != null) {
    return GroupServiceUtil.getGroup(folder.getGroupId());
}
else {
    ThemeDisplay themeDisplay = ....;

    return themeDisplay.getScopeGroup();
}
```
This way, it is more obvious that the different branches are independent and share no mutable state, and that the final return value is 100% determined by each of the branches.

Reply:
I did this change too. Thanks.